sim/midas/src/main/cc/core/simif.h

```diff
@@ -90,25 +103,24 @@ class simif_t {
    * The target cycle is based on the number of clock tokens enqueued
    * (will report a larger number).
    */
-  uint64_t actual_tcycle() { return clock.tcycle(); }
+  uint64_t actual_tcycle() {
+    return registry->get_widget<clockmodule_t>().tcycle();
+  }
 
   /**
    * Provides the current host cycle.
    */
-  uint64_t actual_hcycle() { return clock.hcycle(); }
-
-  /**
-   * Return a reference to the LoadMem widget.
-   */
-  loadmem_t &get_loadmem() { return loadmem; }
+  uint64_t actual_hcycle() {
+    return registry->get_widget<clockmodule_t>().hcycle();
+  }
 
   /// Return the name of the simulated target.
   std::string_view get_target_name() const { return config.target_name; }
 
   /**
-   * Return a reference to the managed stream engine.
+   *  Return the name of the simulated target.
```

Comment:
This doesn't really make sense to me, maybe? 
```suggestion
   *  Return a reference to the registry of all widgets in simulator
```

Reply:
Yikes, got lost in the rebases.